atlasdb-config/src/main/java/com/palantir/atlasdb/factory/Leaders.java

```diff
@@ -148,21 +152,29 @@ public static LocalPaxosServices createInstrumentedLocalServices(
                 PaxosProposerImpl.newProposer(ourLearner, acceptors, learners, config.quorumSize(),
                 leaderUuid, proposerExecutorService));
 
-        InstrumentedExecutorService leaderElectionExecutor = new InstrumentedExecutorService(
-                PTExecutors.newCachedThreadPool(new ThreadFactoryBuilder()
-                        .setNameFormat("atlas-leaders-election-%d")
+        // TODO (jkong): Make the limits configurable.
+        // Current use cases tend to have not more than 10 inflight tasks under normal circumstances.
+        Function<String, ExecutorService> leaderElectionExecutor = (useCase) -> new InstrumentedExecutorService(
+                PTExecutors.newThreadPoolExecutor(
+                        1,
```

Comment:
probably want core pool size to be the number of other potential leaders

Reply:
Yep, that sounds right!